parser/scraper_halooglasi.py

```diff
@@ -54,13 +51,27 @@ def get_apartment_district(self, apartment) -> Optional[str]:
         places = [item.get_text(strip=True) for item in places_items if item.get_text(strip=True)]
         if places:
             return places[2]
-        else:
-            raise ApartmentParseError(f"District is not found for {subtitle_places}")
+        raise ApartmentParseError(f"District is not found for {subtitle_places}")
 
     @suppress(ApartmentParseError, "Price not found")
-    def get_apartment_price(self, apartment) -> Optional[str]:
+    def get_apartment_price(self, apartment) -> Optional[int]:
         price_tag = apartment.find('div', class_='central-feature')
         if price_tag:
-            return price_tag.find('span').text
-        else:
-            raise ApartmentParseError(f"Price is not found for {price_tag}")
+            return int(price_tag.find('span').text.replace('\u00a0', '').replace('€', '').replace('.', '').strip())
+        raise ApartmentParseError(f"Price is not found for {price_tag}")
+
+    @suppress(ApartmentParseError, "Failed to get apartment area")
+    def get_apartment_area(self, apartment) -> Optional[int]:
+        product_features = apartment.find('ul', class_='product-features')
+        features_items = product_features.find_all('li', class_='col-p-1-3')
+        area_div = features_items[0].find('div', class_='value-wrapper')
+        if area_div:
+            return int(area_div.get_text(strip=True).replace('\xa0m2Kvadratura', '').strip())
+        raise ApartmentParseError(f"Apartment area not found for {area_div}")
+
+    @suppress(ApartmentParseError, "Failed to get apartment posting date")
+    def get_apartment_date(self, apartment) -> Optional[str]:
```

Comment:
is it really Optional? .get_text() may return None?

Reply:
I believe the posting date should always be present. Fixed it.